File: modules/dxp/apps/seo-studio/seo-studio-spi/src/main/java/com/liferay/seo/studio/spi/autofix/Autofix.java
/**
 * SPDX-FileCopyrightText: (c) 2026 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.seo.studio.spi.autofix;

/**
 * @author David Truong
 */
public interface Autofix {

	public AutofixResult apply(
			String accessToken, String baseURL,
			String cachedSiteExternalReferenceCode, String pagePath,
			String value)
		throws Exception;

	public String getInsightType();

}

File: modules/dxp/apps/seo-studio/seo-studio-web/src/main/resources/META-INF/resources/js/components/insights_view/types/Autofix.ts
/**
 * SPDX-FileCopyrightText: (c) 2026 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export type AutofixCandidate = {
	rationale?: string;
	value: string;
};

export type ScanInsightItem = {
	externalReferenceCode?: string;
	id: number;
	r_seoStudioPageToSEOStudioScanInsights_seoStudioPage?: {
		pageURL?: string;
		title?: string;
		type?: string;
	};
	state?: number;
};
